src/bases/createWidgetBase.ts

```diff
@@ -117,26 +118,21 @@ const createWidget: WidgetFactory = createStateful
 				const props: VNodeProperties = {};
 
 				this.nodeAttributes.forEach((fn) => {
-					const newProps: VNodeProperties = fn.call(this, assign({}, props));
+					const newProps: VNodeProperties = fn.call(this);
```

Comment:
in both cases, I think it could cause surprising results because it would certainly be order dependent on mixins (which I don't like); It is ported over from the existing functionality so if we think that we need to address this some more, perhaps it would be part of a separate issue?